base/refpointer.jl

```diff
@@ -16,7 +16,19 @@ it can be written `Ref(a, i)` for creating a reference to the `i`-th element of
 When passed as a `ccall` argument (either as a `Ptr` or `Ref` type), a `Ref` object will be
 converted to a native pointer to the data it references.
 
-There is no invalid (NULL) `Ref` in Julia, but a `C_NULL` instance of `Ptr` can be passed to a `ccall` Ref argument.
+There is no invalid (NULL) `Ref` in Julia, but a `C_NULL` instance of `Ptr` can be passed to
+a `ccall` Ref argument.
+
+# Use in broadcasting
+
+Broadcasting with `Ref(x)` treats `x` as a scalar (except when it is a `Ptr`):
```

Comment:
> (except when it is a `Ptr`)

Really?

```
julia> tuple.(Ref(pointer([])), 1:3)
3-element Array{Tuple{Ptr{Any},Int64},1}:
 (Ptr{Any} @0x000000011a3a68f0, 1)
 (Ptr{Any} @0x000000011a3a68f0, 2)
 (Ptr{Any} @0x000000011a3a68f0, 3)
```

Broadcast still treats it like a scalar.  Of course, Ptr's themselves are treated like scalars, so it's a bit of an unnecessary operation, but it just works all the same.

Reply:
I read that in https://docs.julialang.org/en/v1/manual/arrays/#Broadcasting-1,

> Additionally, broadcast is not limited to arrays (see the function documentation), it also handles tuples and treats any argument that is not an array, tuple or Ref (except for Ptr) as a "scalar".

If this is not the case, that part of the docs should be amended, too.

Reply:
Ah! That section is wholly outdated!  Thanks :)

```suggestion
Broadcasting with `Ref(x)` treats `x` as a scalar:
```